File: examples/python/snippets/quickstart-agent.py
import asyncio
import os
from meshagent.api import RoomClient


async def main():
    # Run with:
    # meshagent room connect --room=quickstart --identity=my-user -- python3 quickstart-agent.py
    async with RoomClient() as room:
        path = "sample-agent.document"

        await room.agents.invoke_tool(
            toolkit="meshagent.document-writer",
            tool="run_meshagent.document-writer_task",
            input={
                "path": path,
                "prompt": "write a paragraph about ai and how agents are shaping the future",
            },
        )
        project_id = os.getenv("MESHAGENT_PROJECT_ID")
        if project_id:
            print(
                f"Take a look at it at https://studio.meshagent.com/projects/{project_id}/rooms/{room.room_name}?p={path}"
            )


asyncio.run(main())

File: examples/dart/quickstart-agent.dart
import 'package:meshagent/meshagent.dart';
import 'package:meshagent/helpers.dart' show websocketProtocol;

Future<void> main() async {
  const roomName = 'quickstart';
  const participantName = 'my-user';
  const path = 'sample-agent.document';

  final protocolFactory = websocketProtocol(
    roomName: roomName,
    participantName: participantName,
  );
  final room = RoomClient(protocolFactory: protocolFactory);

  try {
    await room.start();

    await room.agents.call(
      name: 'meshagent.document-writer',
      url: 'https://api.meshagent.life/agents/meshagent.document-writer',
      arguments: const {},
    );

    await room.agents.invokeTool(
      toolkit: 'meshagent.document-writer',
      tool: 'run_meshagent.document-writer_task',
      input: ToolContentInput(
        JsonContent(
          json: {
            'path': path,
            'prompt':
                'write a paragraph about AI and how agents are shaping the future',
          },
        ),
      ),
    );

    print('Take a look at it in MeshAgent Studio.');
  } finally {
    room.dispose();
  }
}
